Clarify Standard Return Codes and Fork Usage

sean-horn opened the pull request:
### Description
If you fork a child process, then a command line capture of the return code will be that from the parent, not the child that does the chef-client run.

sean-horn committed:
Clarify Standard Return Codes and Fork Usage

If you fork a child process, then a command line capture of the return code will be that from the parent, not the child that does the chef-client run.

tas50 commented on chef_master/source/config_rb_client.rst:
```diff
@@ -154,9 +154,10 @@ This configuration file has the following settings:
-   .. note:: The behavior with the default value consists of a warning on the use of deprecated and non-standard exit codes. In a future release of Chef client, using standardized exit codes will be the default behavior.
+   .. note:: The behavior with the default value consists of a warning on the use of deprecated and non-standard exit codes. In the 13.x release of Chef Client and beyond, using standardized exit codes is the default behavior and cannot be changed with this config item.
 
    Changed in Chef Client 12.11 to support standard exit codes.
```
We should remove the mention of Chef 12 at this point. It's EOL and if someone uses it they better be on 12.20

sean-horn committed:
Further clarify client_fork and exit_status usage

More clarity with chef-client versions and client_fork and exit_status settings. Remove Chef Client 12.x statement on standard return codes from @tas50 comment

Commit:
Copyediting

Signed-off-by: Mary Jinglewski <mjinglewski@chef.io>

mjingle reviewed (approved):
![tenor-8868712](https://user-images.githubusercontent.com/5807053/47032376-3c9f8700-d140-11e8-8270-7849634c5049.gif)
